### DISCUSSION TITLE
Implementation of Low Stretch Spanning Tree

### DISCUSSION THREAD

**Discussion by sauravpr** [networkx/networkx#6402] [Algorithms]
Please add functionality for obtaining a _low stretch spanning tree_, both for weighted and unweighted graphs. Two popular algorithms are cited below:

[1] Abraham I, Neiman O. Using petal-decompositions to build a low stretch spanning tree. In Proceedings of the forty-fourth annual ACM symposium on Theory of computing 2012 May 19 (pp. 395-406).

[2] Elkin M, Emek Y, Spielman DA, Teng SH. Lower-stretch spanning trees. In Proceedings of the thirty-seventh annual ACM symposium on Theory of computing 2005 May 22 (pp. 494-503).

**Comment by dschult** [comment]
I have looked into these algorithms since shortly after they were published. Last I checked, even Prof Spielman hasn't implemented them in code that he would want anyone to see. While they are theoretically very powerful in limits of large problems, it is tricky to implement and it is not clear that the constants in the asymptotic limits are small enough to make them effective. But I haven't looked into this in over 5 years, so things may have changed.

Are you able to point to any resources that indicate this is possible or reasonable to do? The papers are actually fairly old, as CS papers go. Are there more recent follow-up papers?  Would you be interested in investing the time to implement them?

**Reply by sauravpr** [reply]
Thank you Prof. Schult for your response! You are right, the implementation of Prof Spielman's paper is not available in his GitHub repo either, but he has provided a variant of the prior low stretch algorithm (https://docs.juliahub.com/Laplacians/poVbr/1.1.1/LSST/). I am very new to this topic, so am not very familiar with current papers. I will let you know if I find something recent. However, I did find one implementation of Prof. Spielman's paper below, https://github.com/vardhinibhasker/LowStretchSpanningTree
but I have not run it myself yet. Unfortunately, I am currently unable to work on implementing them from scratch.
